analytics.py

```diff
@@ -0,0 +1,219 @@
+import math
+ import random
+ from utils import euclidean_distance, n_random_points
+
+ def find_largest_city(gj):
+     """
+     Iterate through a geojson feature collection and
+     find the largest city.  Assume that the key
+     to access the maximum population is 'pop_max'.
+
+     Parameters
+     ----------
+     gj : dict
+          A GeoJSON file read in as a Python dictionary
+
+     Returns
+     -------
+     city : str
+            The largest city
+
+     population : int
+                  The population of the largest city
+     """
+
+     featureList = gj['features']
+
+     max_population = 0
+     for featureEntry in featureList:
+         if featureEntry["properties"]["pop_max"] > max_population:
+             max_population = featureEntry["properties"]["pop_max"]
+             city = featureEntry["properties"]["nameascii"]
+
+
+     return city, max_population
+
+ def write_your_own(gj):
+
+
+ def mean_center(points):
+     """
+     Given a set of points, compute the mean center
+
+     Parameters
+     ----------
+     points : list
+          A list of points in the form (x,y)
+
+     Returns
+     -------
+     x : float
+         Mean x coordinate
+
+     y : float
+         Mean y coordinate
+     """
+
+
+     sums = map(sum,zip(*points))
+     sumsL = list(sums)
+     avgs = map(lambda xy: xy/len(points),sumsL)
+     avgsL = list(avgs)
+     x = avgsL[0]
+     y = avgsL[1]
+
```

Comment:
This is spot on.  Do you understand what is going on here?  This is a style that we will not even look at for a few more weeks.

Another solution might be:

```python
sum_x = 0
sum_y = 0
for p in points:
    sum_x = sum_x + p[0]
    sum_y = sum_y + p[1]

mean_x = sum_x / len(points)
mean_y = sum_y / len(points)  
```